sdk/storage/storage-file-datalake/review/storage-file-datalake.api.md

```diff
@@ -0,0 +1,1424 @@
+    // (undocumented)
+    copySource?: string;
+    // (undocumented)
+    copyStatus?: CopyStatusType;
+    // (undocumented)
+    copyStatusDescription?: string;
+    // (undocumented)
+    createdOn?: Date;
+    // (undocumented)
+    date?: Date;
+    // (undocumented)
+    destinationSnapshot?: string;
+    // (undocumented)
+    encryptionKeySha256?: string;
+    // (undocumented)
+    etag?: string;
+    // (undocumented)
+    isIncrementalCopy?: boolean;
+    // (undocumented)
+    isServerEncrypted?: boolean;
+    // (undocumented)
+    lastModified?: Date;
+    // (undocumented)
+    leaseDuration?: LeaseDurationType;
+    // (undocumented)
+    leaseState?: LeaseStateType;
+    // (undocumented)
+    leaseStatus?: LeaseStatusType;
+    // (undocumented)
+    metadata?: Metadata;
+    // (undocumented)
+    requestId?: string;
+    // (undocumented)
+    version?: string;
+}
+
+// @public (undocumented)
+export interface PathGetPropertiesOptions extends CommonOptions {
+    // (undocumented)
+    abortSignal?: AbortSignalLike;
+    // (undocumented)
+    conditions?: DataLakeRequestConditions;
+}
+
+// @public (undocumented)
+export type PathGetPropertiesResponse = PathGetPropertiesHeaders & {
+    _response: HttpResponse & {
+        parsedHeaders: PathGetPropertiesHeaders;
+    };
+};
+
+// @public (undocumented)
+export interface PathHttpHeaders {
+    // (undocumented)
+    cacheControl?: string;
+    // (undocumented)
+    contentDisposition?: string;
+    // (undocumented)
+    contentEncoding?: string;
+    // (undocumented)
+    contentLanguage?: string;
+    // (undocumented)
+    contentMD5?: Uint8Array;
+    // (undocumented)
+    contentType?: string;
+}
+
+// @public
+export interface PathList {
+    // (undocumented)
+    pathItems?: Path[];
+}
+
+// @public (undocumented)
+export interface PathMoveOptions extends CommonOptions {
+    // (undocumented)
+    abortSignal?: AbortSignalLike;
+    // (undocumented)
+    conditions?: DataLakeRequestConditions;
+    // (undocumented)
+    destinationConditions?: DataLakeRequestConditions;
+}
+
+// @public (undocumented)
+export type PathMoveResponse = PathRemoveHeaders & {
+    _response: HttpResponse & {
+        parsedHeaders: PathRemoveHeaders;
+    };
+};
+
+// @public (undocumented)
+export interface PathPermissions {
+    // (undocumented)
+    extendedAcls: boolean;
+    // (undocumented)
+    group: RolePermissions;
+    // (undocumented)
+    other: RolePermissions;
+    // (undocumented)
+    owner: RolePermissions;
+    // (undocumented)
+    stickyBit: boolean;
+}
+
+// @public (undocumented)
+export interface PathRemoveHeaders {
+    // (undocumented)
+    contentLength?: number;
+    // (undocumented)
+    date?: Date;
+    // (undocumented)
+    eTag?: string;
+    // (undocumented)
+    lastModified?: Date;
+    // (undocumented)
+    requestId?: string;
+    // (undocumented)
+    version?: string;
+}
+
+// @public
+export enum PathRenameMode {
```

Comment:
Do we need the enum here and below, or can we get away with string literal types instead?

Reply:
This type should be internal, will exclude from the export list. 

While below `PathResourceType` should be exported.

Any concerns for the enum type here?

Reply:
If it's internal, no concerns with enum.

For external, I'd prefer string literal type unions unless it's clearly better to export an enum. Since it's just simple strings, I think almost all developers would prefer `foo("directory")` over `import { PathResourceType } from "@azure/storage-file-datalake"; foo(PathResourceType.directory)`.

Reply:
If am correct, string enum should compatible with following 2 usage:

`foo("directory")` or `foo(PathResourceType.directory)`

Concern for string union type, customers need to check API reference document to get full list candidate values, in a dev env without intelligence.